src/middlewared/middlewared/main.py

```diff
@@ -681,6 +687,23 @@ def _call_prepare(
             if methodobj._pass_app['message_id']:
                 args.append(message_id)
 
+        is_coroutine = asyncio.iscoroutinefunction(methodobj)
+        if (
+            serviceobj._config.namespace in ('zfs.resource')
```

Comment:
This looks ambiguous to a human reader. Did you intend it to work like this?
```
>>> 'zfs.res' in ('zfs.resource')
True
```

Reply:
No, there will be a `zpool.resource` namespace eventually and so I was putting it in a tuple of strings for this fact.

Reply:
I could probably just remove this check and make it so the `class Config` object can accept a boolean to keep it simple.

Reply:
I agree. Explicit is better than implicit.